hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecordPayload.java

```diff
@@ -41,9 +41,19 @@
    * When more than one HoodieRecord have the same HoodieKey, this function combines them before attempting to
    * insert/upsert (if combining turned on in HoodieClientConfig).
    */
-  @PublicAPIMethod(maturity = ApiMaturityLevel.STABLE)
+  @Deprecated
   T preCombine(T another);
 
+  /**
+   * When more than one HoodieRecord have the same HoodieKey, this function combines all fields(which is not null)
+   * before attempting to insert/upsert (if combining turned on in HoodieClientConfig).
+   *
+   */
+  @PublicAPIMethod(maturity = ApiMaturityLevel.STABLE)
+  default T preCombine(T another, Schema schema) throws IOException {
```

Comment:
@bvaradar is exploring if we can just use `combineAndGetUpdateValue()` for preCombine as well. but I guess, preCombine works with the payload, not avro records. 
WOuld be good to get his inputs as well